app/2.0/docs/devguide/dom-template.md

```diff
@@ -38,15 +55,159 @@ Example: { .caption }
 
 In this example, the DOM template and the JavaScript that defines the element are in the same file. You can also put these portions in separate files, but the DOM template must be parsed before the element is defined.
 
-
 **Note:** Elements should generally be defined outside of the main document,
 except for testing. For caveats about defining elements in the main document,
 see [main document definitions](registering-elements#main-document-definitions).
 {.alert .alert-info}
 
-## Automatic node finding {#node-finding}
+### Specify a string template
+
+As an alternative to specifying the element's template in markup, you can specify a string template
+by creating a static `template` getter that returns a string.
+
+This getter is called _once_, when the first instance of the element is upgraded.
+
+```js
+class MyElement extends Polymer.Element {
+
+  static get template() {
+    return `<style>:host { color: blue; }</style>
+       <h2>String template</h2>
+       <div>I've got a string template!</div>`
+  }
+}
+customElements.define('my-element', MyElement);
+```
+
+When using a string template, the element doesn't need to provide an `is` getter (however, the tag
+name still needs to  be  passed as the first argument to `customElements.define`).
+
+### Inherited templates
+
+An element that extends another Polymer element can inherit its template. If the element doesn't
+provide its own DOM template (using either a `<dom-module>` or a string template), Polymer uses the
+same template as the superclass, if any.
+
+You can also modify a superclass template by defining a `template` getter that returns a modified
+template element. If you're going to modify the superclass template, there are a couple of
+important rules:
+
+-   Don't modify the superclass template in place; make a copy before modifying.
+-   If you're doing anything expensive, like copying or modifying an exisitng the template,
+    you should memoize the modified template so you don't have to regenerate it when the
+    getter is called.
+
+The following example shows a simple modification of a parent template:
+
+```js
+(function() {
+  let memoizedTemplate;
+
+  class MyExtension extends MySuperClass {
+    static get template() {
+      if (!memoizedTemplate) {
+        // retrieve superclass template
+        memoizedTemplate = MySuperClass.template;
+        // add a node to the template.
+        let div = document.createElement('div');
+        div.textContent = 'Hello from an extended template.'
+        memoizedTemplate.content.appendChild(div);
+      }
+      return memoizedTemplate;
+    }
+  }
+
+})();
+```
+
+The following example shows a more complicated could wrap a superclass template with a child
```

Comment:
Hah! That'll teach me to submit stuff on Monday that I was editing on Friday 😊.

Perhaps the sentence is complicated enough without "complicated" and could just be: "The following example shows how an element could wrap a superclass template with a child template."